src/schemathesis/specs/openapi/schemas.py

```diff
@@ -479,6 +479,10 @@ def validate_response(self, operation: APIOperation, response: GenericResponse)
             return
         try:
             if isinstance(response, requests.Response):
+
+                if response.encoding == 'utf-8' and (response.text[0:1] in ('\ufeff', '\ufffe')): # utf-8 with BOM
```

Comment:
Could we have the byte order mark char as module-level constant instead? Btw, why there are two of them?

Also, a comment about the motivation would be nice. E.g. that RFC 7159 [forbids](https://www.rfc-editor.org/rfc/rfc7159#section-8.1) BOM, but we need this to workaround some non-compliant servers.

Reply:
Thank you for your attention to this question and interesting links about it!

> Could we have the byte order mark char as module-level constant instead? 
Of course!

> Btw, why there are two of them?
As mentioned in [wikipedia BOM article](https://en.wikipedia.org/wiki/Byte_order_mark):

```
 * If the 16-bit units are represented in [big-endian](https://en.wikipedia.org/wiki/Big-endian) byte order, the BOM will appear in the sequence of bytes as **0xFE 0xFF**
  * If the 16-bit units use [little-endian](https://en.wikipedia.org/wiki/Little-endian) order, the BOM will appear in the sequence of bytes as **0xFF 0xFE**
```
On my intel x64 box those bytes appears as '\ufeff', but maybe on other platforms it might be '\ufffe' in that situation, therefore I'd add this.

Approach proposed in `requests` PR (use to all repsonses in utf-8 python's utf-8-sig encoding to decode) also works good  in my situation, I have tested. As I read after on [python codecs docs](https://docs.python.org/3/library/codecs.html#encodings-and-unicode): "On decoding utf-8-sig will skip those three bytes if they appear as the first three bytes in the file.", but if utf-8 have no BOM, 'utf-8-sig' acts linke 'utf-8' encoding, I can confirm it with my tests :)

I trust you to choose best solution! Thank you in advance!

Reply:
Hmm, the article section you are referring to is about UTF-16, right? Here we have UTF-8, but I am not sure if `\ufffe` is relevant to it

Reply:
Though, it could be tested :) I'll check it out

Reply:
> Hmm, the article section you are referring to is about UTF-16, right? Here we have UTF-8, but I am not sure if `\ufffe` is relevant to it

Well, as I'd investigated, BOM in UTF-8 is 0xEF 0xBB 0xBF bytes, and that BOM not about byte order, it's about signaling that it's UTF-8 data (it so happened). But, when `requests` receive raw http output, it converts received bytes to internal python string representation through decoding raw data with 'utf-8' encoder. Python internal representation according to [PEP 393](https://peps.python.org/pep-0393/) may be LATIN1, UCS2 (UTF-16) or UCS4 (UTF-32).

In my case python's internal representantion was USC2/UTF-16, thus on my box 0xEF 0xBB 0xBF bytes of UTF-8 becomes `\ufffe` in python's string.

As far as I dive in this, `requests` PR solution to use 'utf-8-sig' to decode 'utf-8' seems be better solution. Sorry if I confused at all.